### PR TITLE
[WIP] Integrating UX docs more closely into Symfony docs

### PR DESCRIPTION
**Pull request by weaverryan**
Hi!

Partner to https://github.com/symfony/ux/pull/225

TODO

* [x] Build this locally and proofread
* [x] Wait for https://github.com/symfony/ux/pull/225 to be merged and bundle docs to be put onto symfony.com

### CODE REVIEW

File: frontend/ux.rst [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+-------------------------------
+
+Because Stimulus is used by developers outside of Symfony, many tools
+exist beyond the UX packages:
+
+-   `stimulus-use`_: Add composable behaviors to your Stimulus controllers, like
+    debouncing, detecting outside clicks and many other things.
+
+-   `stimulus-components`_ A large number of pre-made Stimulus controllers, like for
+    Copying to clipboard, Sortable, Popover (similar to tooltips) and much more.
```
Comment (wouterj) [inline frontend/ux.rst]:
```suggestion
* `stimulus-use`_: Add composable behaviors to your Stimulus controllers, like
  debouncing, detecting outside clicks and many other things.

* `stimulus-components`_ A large number of pre-made Stimulus controllers, like for
  Copying to clipboard, Sortable, Popover (similar to tooltips) and much more.
```

Not sure if it makes a difference, but let's keep things as simple as possible (least chances of rst-parser not understanding it correctly)

File: templates.rst [hunk 2]
```diff
@@ -408,6 +409,22 @@ gives you access to these variables:
...
+Twig components are an alternative way to render templates, where each template
+is bound to a "component class". This makes it easier to render and re-use
+small template "units" - like an alert, markup for a modal, or a category sidebar.
+
+For more information, see `UX Twig Component`_.
+
+Twig components also have one other superpower: they can become "live", where
+they automatically update (via Ajax) as the user interacts with them. For example,
+when your user types into a box, your Twig component will re-render via Ajax to
+show a list of results. It's... pretty cool.
```
Comment (wouterj) [inline templates.rst]:
```suggestion
show a list of results.
```

It's fun, but a bit too informal compared to the rest of the docs I think ;)

**Review by wouterj — APPROVED** [review]
This is great, thanks Ryan! The stream of awesome contributions from you just doesn't stop the past months :)

Two other points:
* Let's update the "Symfony UX initiative" link in the form type reference to point to `/frontend/ux`
* Please rebase this PR to remove the merge commit

**Comment by weaverryan** [comment]
Ready to go again :)

**Comment by wouterj** [comment]
Thanks @weaverryan!

Is there a reason to target 5.4 here, instead of 5.3 or 4.4? Afaik, UX is available in all versions :)

**Comment by weaverryan** [comment]
Just that we, I think, typically target the "current Symfony version" for new docs, right? To avoid extra pain merging up from old branches? But certainly, I could target something older. WDYT?

**Comment by javiereguiluz** [comment]
Ryan, I understand what you say ... but we always merge UX things in 4.4 because they are disconnected from Symfony versioning. Note that if, for any reason, it's complicated for us to write these docs for 4.4, we can change this practice. Thanks!

**Comment by wouterj** [comment]
Note that I wanted to merge this in 5.3 yesterday (4.4 gave lots of conflicts) and then discovered that Turbo and "Stimulus & Symfony UX" also wasn't merged in 5.3.

As far as I know, we've only applied the practice to only update the latest version for a very brief time and are updating older versions again.

EDIT: removed the demotivating text as you've updated this PR to target 4.4 already :rocket:  :)

**Comment by weaverryan** [comment]
No problem - it seems very pragmatic :). 

As you mentioned, some things didn't exist on 4.4. So `simple-example.rst` is now a copy of what I had proposed on 5.4, so it also contains a bunch of unrelated changes.

But, this should be ready I think :)

**Pull request merged**